tests/tensorflow/test_tensorflow_autolog.py

```diff
@@ -0,0 +1,213 @@
+# pep8: disable=E501
+
+from __future__ import print_function
+
+import collections
+import shutil
+import pytest
+import tempfile
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from tensorflow.python.keras import layers
+
+import mlflow
+import mlflow.tensorflow
+
+SavedModelInfo = collections.namedtuple(
+        "SavedModelInfo",
+        ["path", "meta_graph_tags", "signature_def_key", "inference_df", "expected_results_df"])
+
+
+@pytest.fixture
+def tf_keras_random_data():
+    mlflow.tensorflow.autolog(metrics_every_n_steps=2)
+    def random_one_hot_labels(shape):
+        n, n_class = shape
+        classes = np.random.randint(0, n_class, n)
+        labels = np.zeros((n, n_class))
+        labels[np.arange(n), classes] = 1
+        return labels
+
+    data = np.random.random((1000, 32))
+    labels = random_one_hot_labels((1000, 10))
+
+    model = tf.keras.Sequential()
+
+    model.add(layers.Dense(64, activation='relu'))
+    model.add(layers.Dense(64, activation='relu'))
+    model.add(layers.Dense(10, activation='softmax'))
+
+    model.compile(optimizer=tf.train.AdamOptimizer(0.001),
+                  loss='categorical_crossentropy',
+                  metrics=['accuracy'])
+    run = mlflow.start_run()
+    model.fit(data, labels, epochs=10)
+    mlflow.end_run()
+    return run.info.run_id
+
+
+@pytest.fixture
+def get_tf_keras_random_data_run_instance(tf_keras_random_data):
+    client = mlflow.tracking.MlflowClient()
+    run = client.get_run(tf_keras_random_data)
+    return run
+
+
+@pytest.mark.small
+def test_tf_keras_autolog_logs_epoch_acc(get_tf_keras_random_data_run_instance):
+    assert 'epoch_acc' in get_tf_keras_random_data_run_instance.data.metrics
```

Comment:
For the metrics tests, it probably doesn't make sense to assert on the value (since the values may change with each test run), but maybe we can assert that metrics are in fact logged every `n` steps (i.e. assert on the `step` of the logged metrics)?

Reply:
Also, it'd be nice to have test cases for tf.Keras, tf.estimator, TF core that don't pass `metrics_every_n_steps` to `autolog` and verify that metrics are logged every 100 steps in the default case. Let's think about whether there are ways to do this without duplicating a bunch of test code (i.e. we can probably have a helper function that trains a model after calling autolog with the specified value of `metrics_every_n_steps`, and call that helper function in our [fixture](https://github.com/mlflow/mlflow/pull/1520/files/39836b999585ed1dff17dd9581779d31e911a56e..bced27e3fd473d578f5481759596a26d719e0a0b#diff-a00285f1bcbcd92a659c4b4fc1efa871R24)). It'd also be good to test that if we call `mlflow.tensorflow.autolog` twice with different values of `metrics_every_n_steps`, the second value takes precedence.